corehq/apps/ota/utils.py

```diff
@@ -165,3 +168,21 @@ def get_restore_user(domain, couch_user, as_user):
         restore_user = couch_user.to_ota_restore_user(domain)
 
     return restore_user
+
+
+def handle_401_response(f):
+    def _inner(request, domain, *args, **kwargs):
+        auth_type = determine_authtype_from_request(request)
+        response = f(request, domain, *args, **kwargs)
```

Comment:
Well we need to act only if auth_type is basic for now. Would result in a new `if` block if not combined with the existing check.... but now when i see this i guess yup! lets move that inside block.